### ISSUE TITLE
Provide built-in SFTP support, discourage FTP

### ISSUE THREAD

**Issue by Gallaecio** [scrapy/scrapy#4180]
Scrapy provides built-in support for exporting data through FTP. [FTP is considered insecure](https://bandit.readthedocs.io/en/latest/blacklists/blacklist_imports.html#b402-import-ftplib).

We should consider:
- Adding built-in support for SFTP; see https://github.com/scrapy-plugins/scrapy-feedexporter-sftp
- Discourage FTP usage (through documentation and a warning?)
- Deprecate FTP support?

**Comment by nyov** [comment]
I agree with (re)moving the ftp exporter to a scrapy-plugin repo, and getting rid of `utils/ftp.py`. I'm not sure we should pull sftp (and paramiko) into scrapy proper, though.

**Comment by mxdev88** [comment]
FTP is still useful.  Could it considered adding an option to secure the connection with [FTPS](https://docs.python.org/3.6/library/ftplib.html?highlight=ftp#ftplib.FTP_TLS) instead of removing?

**Comment by further-reading** [comment]
Would adding a sftp download handler too be within scope for this issue? We currently support downloads from ftp uris but not sftp. I've had to make a hacky custom handler for it and I'd be happy to improve it and fold it into scrapy core if this seems like a valuable feature.

**Comment by Gallaecio** [comment]
> I'm not sure we should pull sftp (and paramiko) into scrapy proper, though.

If the concern is paramiko specifically, we could handle this like we do with Amazon Web Services S3 support, making the dependency optional, only needed when wishing to use this feature.
